### PR TITLE
reaver: database on readwrite medium

### LINKED ISSUE #25297: reaver-wps and reaver-wps-t6x: CONF_DIR should point to writable medium, not into /nix/store
## Issue description

compile-time define ```CONF_DIR``` points to ```/nix/store/....reaver.../etc```
The directory is used to hold sqlite database and saved sessions, so it should point to writable medium, better not on ```tmpfs```. Something in ```/etc``` or ```/var```.

> this moment is unclear to me (so I created an issue, not a PR): could a nix package, not a nixos module, create and maintain its own directory in ```/etc``` or ```/var```?. Technically yes (this software is always run by ```root```), but so far I have seen no such example. The question is about policy.

The current setup does not prevent ```reaver``` to start (although it prevent to start ```wash```).

### Steps to reproduce

Install ```reaver```, start ```strace wash -i your-wireless-device```. See that it tries to open sqlite database for writing. Then it fails becaue the database is on readonly ```/nix/store```. So it tries forever.

## Technical details

* System: 17.09.git.bd2d225 (Hummingbird) 
* Nix version: nix-env (Nix) 1.12pre5152_915f62fa
* Nixpkgs version: "17.09.git.bd2d22589a"

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

Fixes #25297

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/networking/reaver-wps-t6x/default.nix [hunk 1]
```diff
@@ -13,9 +14,23 @@ stdenv.mkDerivation rec {
...
+  configureFlags = "--sysconfdir=${confdir}";
+
+  installPhase = ''
+    mkdir -p $out/{bin,etc,share}
+    cp reaver.db $out/etc/
+
+    for prog in reaver wash; do
+      cp $prog $out/share/
+      cat > $out/bin/$prog <<EOF
+    [ -f ${confdir}/reaver/reaver.db ] || (mkdir -p ${confdir}/reaver; cp $out/etc/reaver.db ${confdir}/reaver/)
```
Comment (Mic92) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: why is a conditional check here?

File: pkgs/tools/networking/reaver-wps-t6x/default.nix [hunk 2]
```diff
@@ -3,6 +3,7 @@
 stdenv.mkDerivation rec {
   version = "1.5.2";
   name = "reaver-wps-t6x-${version}";
+  confdir = "/var/db/${name}"; # the sqlite database is at "${confdir}/reaver/reaver.db"
```
Comment (Mic92) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: this is an impure check.

File: pkgs/tools/networking/reaver-wps-t6x/default.nix (hunk 1 above)
Comment (Mic92) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: and here files are installed outside of nix store

Comment (volth) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: No, it is not installer code, it is wrapper code in ```$out/bin/reaver``` in ```$out/bin/wash```

File: pkgs/tools/networking/reaver-wps-t6x/default.nix (hunk 2 above)
Comment (volth) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: a check?

File: pkgs/tools/networking/reaver-wps-t6x/default.nix (hunk 1 above)
Comment (Mic92) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: sorry I misread the code.

Comment (Mic92) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: there is a shebang missing in the script and `|| (mkdir -p ${confdir}/reaver; cp $out/etc/reaver.db ${confdir}/reaver/)`
can be replaced by `|| install -D ${confdir}/reaver $out/etc/reaver.db`

File: pkgs/tools/networking/reaver-wps-t6x/default.nix (hunk 2 above)
Comment (Mic92) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: see below

File: pkgs/tools/networking/reaver-wps-t6x/default.nix (hunk 1 above)
Comment (Mic92) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: btw. `makeWrapper` also have a run argument, which can be used for the purpose here.

Comment (volth) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: ```install``` cannot fix the case when the directory ```/var/db/reaver-wps-1.4/reaver``` exists but the file ```/var/db/reaver-wps-1.4/reaver/reaver.db``` does not (I just hit this problem).

Comment (Mic92) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: I meant: `|| install -d $out/etc/reaver.db ${confdir}/reaver/reaver.db`

Comment (volth) [inline pkgs/tools/networking/reaver-wps-t6x/default.nix]: this works

**Comment by joachifm** [comment]
Is this ready?

**Comment by volth** [comment]
seems so

**Pull request merged**